src/components/Button/Button.tsx

```diff
@@ -30,8 +30,17 @@ const Button: React.FC<ButtonProps> = React.forwardRef(
 			`}
 			aria-busy={!!loading}
 		>
-			{loading && <Loading className="btn__loading" name={icon} aria-hidden />}
-			{!loading && icon && <Icon className="btn__icon" name={icon} />}
+			{loading && <Loading className="btn__loading btn__icon" name={icon} aria-hidden />}
+			{!loading &&
+				icon &&
+				(typeof icon === 'string' ? (
+					<Icon className="btn__icon" name={icon} />
+				) : (
+					React.cloneElement(icon, {
+						...icon.props,
```

Comment:
You can remove that, `React.cloneElement` already merges the element's props with the ones you pass.